### DISCUSSION TITLE
How to use Portal in remix ?

### DISCUSSION THREAD

**Discussion by WreewanMorhee** [remix-run/remix#7874] [Q&A] [answered]
in SPA, I use library: react-portal to make portal
in Remix, it will crash
and also, I cannot find `createPortal` this function in Remix

so 
How could I use Portal in Remix ?

**Comment by sergiodxa** [comment]
You need to use it only client side, you can’t server-side render a portal.

**Reply by WreewanMorhee** [reply]
wow !! ..amazing fact...
any reference I could read or the reason why ?

**Reply by kiliman** [reply, marked as answer]
Portals require an actual DOM element as the root. SSR doesn't create DOM elements, just HTML as a string.

**Reply by kiliman** [reply]
If you need to use `createPortal`, just wrap it inside `<ClientOnly>` from `remix-utils`
```ts
<ClientOnly>
  {() => createPortal(<Modal />, document.getElementById('modal-portal')}
</ClientOnly>
```

**Reply by WreewanMorhee** [reply]
Thank u two !

**Reply by theocerutti** [reply]
I understand that it's not possible to use createPortal on server-side but... it there any ways to trick it ?
If it's really not possible what is the recommended way to create drawer using server-side rendering? (currenlty we use hash url so it can be server-side rendered but it is really limiting...)
